cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+	if config.DeployType == "blue-green" {
+		return "", fmt.Errorf("Blue-green plugin requires app name to be passed (see https://github.com/bluemixgaragelondon/cf-blue-green-deploy/issues/27)")
+	}
+	if len(config.Manifest) == 0 {
+		return "", fmt.Errorf("Manifest file not provided in configuration. Cannot retrieve app name")
+	}
+	fileExists, err := fileUtils.FileExists(config.Manifest)
+	if err != nil {
+		return "", err
+	}
+	if !fileExists {
+		return "", fmt.Errorf("Manifest file '%s' not found. Cannot retrieve app name", config.Manifest)
+	}
+	m, err := _getManifest(config.Manifest)
+	if err != nil {
+		return "", err
+	}
+	apps, err := m.GetApplications()
+	if err != nil {
+		return "", err
+	}
+	if len(apps) > 0 {
+		namePropertyExists, err := m.ApplicationHasProperty(0, "name")
+		if err != nil {
+			return "", err
+		}
+		if namePropertyExists {
+			appName, err := m.GetApplicationProperty(0, "name")
+			if err != nil {
+				return "", err
+			}
+			if name, ok := appName.(string); ok {
+				if len(name) > 0 {
+					return name, nil
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("No appName available in manifest '%s'", config.Manifest)
+}
+
+func handleSmokeTestScript(smokeTestScript string) ([]string, error) {
+
+	if smokeTestScript == "blueGreenCheckScript.sh" {
+		// what should we do if there is already a script with the given name? Should we really overwrite ...
+		err := fileUtils.FileWrite(smokeTestScript, []byte(defaultSmokeTestScript), 0755)
+		if err != nil {
+			return []string{}, err
+		}
+		log.Entry().Debugf("smoke test script '%s' has been written.", smokeTestScript)
+	}
+
+	if len(smokeTestScript) > 0 {
+		err := fileUtils.Chmod(smokeTestScript, 0755)
+		if err != nil {
+			return []string{}, err
+		}
+		pwd, err := fileUtils.Getwd()
+		if err != nil {
+			return []string{}, err
+		}
+
+		return []string{"--smoke-test", fmt.Sprintf("%s/%s", pwd, smokeTestScript)}, nil
+	}
+	return []string{}, nil
+}
+
+func prepareBlueGreenCfNativeDeploy(config *cloudFoundryDeployOptions) (string, []string, []string, error) {
+
+	smokeTest, err := handleSmokeTestScript(config.SmokeTestScript)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+
+	var deployOptions = []string{}
+
+	if !config.KeepOldInstance {
+		deployOptions = append(deployOptions, "--delete-old-apps")
+	}
+
+	if len(config.Manifest) > 0 {
+		manifestFileExists, err := fileUtils.FileExists(config.Manifest)
+		if err != nil {
+			return "", []string{}, []string{}, err
+		}
+
+		if !manifestFileExists {
+
+			log.Entry().Infof("Manifest file '%s' does not exist", config.Manifest)
+
+		} else {
+
+			manifestVariables, err := toStringInterfaceMap(toParameterMap(config.ManifestVariables))
+			if err != nil {
+				return "", []string{}, []string{}, err
+			}
+
+			manifestVariablesFiles, err := removeDefaultManifestVariableFileIfItIsTheOnlyFileAndThatFileDoesNotExist(config.ManifestVariablesFiles)
+			if err != nil {
+				return "", []string{}, []string{}, err
```

Comment:
Should be wrapped. The only error which can come from `removeDefaultManifestVariableFileIfItIsTheOnlyFileAndThatFileDoesNotExist` is if it is unable to check for the existence of the default variables files. It will be completely obscure. It would be equally OK to not return an error from `removeDefaultManifestVariableFileIfItIsTheOnlyFileAndThatFileDoesNotExist` (there, I typed that twice in the same comment...) at all.

Reply:
done